staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -234,6 +236,14 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return healthCheck()
 	}))
 
+	readyCheck, err := storagefactory.CreateReadyCheck(s.StorageConfig, c.DrainedNotify())
+	if err != nil {
+		return err
+	}
+	c.ReadyzChecks = append(c.ReadyzChecks, healthz.NamedCheck("etcd-readiness", func(r *http.Request) error {
```

Comment:
we have a method for this

Reply:
I can only see AddHealthChecks which adds healthcheck to readyz/livez/healthz. It even has `We should prefer this to adding healthChecks directly to the config unless we explicitly want to add a healthcheck only to a specific health endpoint.` in its description. Am I missing something?

Reply:
Here is the function that @sttts was talking about:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/healthz.go#L63

But that is the function at the generic server level, whereas for etcd checks we're operating still at config level.

That said - instead of doing it manually, please create a AddReadyzCheck to the Config struct.